grafana-plugin/src/containers/MobileAppVerification/MobileAppVerification.tsx

```diff
@@ -76,33 +98,63 @@ const MobileAppVerification = observer(({ userPk }: Props) => {
     content = <Text type="primary">{errorFetchingQRCode || errorDisconnectingMobileApp}</Text>;
   } else if (mobileAppIsCurrentlyConnected) {
     content = (
-      <VerticalGroup>
-        <Text type="primary">Your mobile app is currently connected. Click below to disconnect.</Text>
-        <DisconnectButton onClick={disconnectMobileApp} />
+      <VerticalGroup spacing="lg">
+        <Text strong type="primary">
+          App connected <Icon name="check-circle" size="md" className={cx('icon')} />
+        </Text>
+        <Text type="primary">
+          You can sync one application to your account. To setup new device please disconnect app first.
+        </Text>
+        <div className={cx('disconnect__container')}>
+          <img src={qrCodeImage} className={cx('disconnect__qrCode')} />
+          <DisconnectButton onClick={disconnectMobileApp} qrCode={QRCodeValue} />
+        </div>
       </VerticalGroup>
     );
   } else if (QRCodeValue) {
     content = (
-      <VerticalGroup>
-        <QRCode value={QRCodeValue} />
-        <Text type="primary">
-          Note: the QR code is only valid for one minute. If you have issues connecting your mobile app, try refreshing
-          this page to generate a new code.
+      <VerticalGroup spacing="lg">
+        <Text type="primary" strong>
+          Sign In
+        </Text>
+        <Text type="primary">Open Grafana IRM mobile application and scan this code to sync it with your account.</Text>
+        <div className="u-width-100 u-flex u-flex-center">
+          <QRCode value={QRCodeValue} />
+        </div>
+        <Text type="primary" className="u-break-word">
+          <strong>Note:</strong> the QR code is only valid for one minute. If you have issues connecting your mobile
+          app, try refreshing this page to generate a new code.
         </Text>
       </VerticalGroup>
     );
   }
 
   return (
-    <HorizontalGroup>
-      <Block bordered withBackground>
-        {content}
-      </Block>
-      <Block bordered withBackground>
+    <div className={cx('container')}>
+      <Block shadowed bordered withBackground className={cx('container__box')}>
         <DownloadIcons />
       </Block>
-    </HorizontalGroup>
+      <Block shadowed bordered withBackground className={cx('container__box')}>
+        {content}
+      </Block>
+    </div>
   );
+
+  function isUserConnected(user?: User): boolean {
+    return !!(user || userStore.currentUser).messaging_backends[BACKEND]?.connected;
+  }
+
+  async function pollUserProfile(): Promise<void> {
+    clearInterval(userIntervalId);
+    setUserIntervalId(undefined);
+
+    const user = await userStore.loadUser(userPk);
+    if (!isUserConnected(user)) {
+      setUserIntervalId(setInterval(() => pollUserProfile(), INTERVAL_MS));
+    } else {
+      setMobileAppIsCurrentlyConnected(true);
+    }
+  }
```

Comment:
nitpick; I would suggest moving these two functions towards the top of `MobileAppVerification` before they are referenced. Also is it beneficial to wrap these in a `useCallback`?

Reply:
I think `useCallback` is overused in this project, I don't see a reason why it should be used in this place. There's nothing expensive about that function.